### ISSUE TITLE
Build server-only bundle and skip client-bundle

### ISSUE THREAD

**Issue by kevcodez** [nuxt/nuxt.js#6860]
### What problem does this feature solve?

I have a specific use-case, where I only render the pages on the server-side (using Nuxt programatically).

The build time would greatly decrease, if I could just skip building the client-bundle entirely.

<div align="right"><sub><em>This feature request is available on <a href="https://cmty.app/nuxt">Nuxt</a> community (<a href="https://cmty.app/nuxt/nuxt.js/issues/c10195">#c10195</a>)</em></sub></div>

**Comment by Atinux** [comment]
Hi @kevcodez 

So we agree that you don't need JavaScript at all on your generated pages?

**Comment by kevcodez** [comment]
@Atinux Yes. There is no further navigation or whatsoever. Let me elaborate on my specific use-case.

I have an express API, receiving a POST request with some data

```json
{
"template": "invoice",
"data": {
"firstName": "foo",
"lastName": "bar"
}
```

The endpoint then uses `nuxt.renderRoute` to dynamically render a page (

```js

// Nuxt and express initialization

app.post('/generate', async (req, res, next) => {
  await nuxt.ready()
  const jsonRequest = req.body
  const route = '/' + jsonRequest.template
  const {
    html,
    _error,
    _redirected
  } = await nuxt.renderRoute(route, {
    data: jsonRequest.data
  })

  res.set('Content-Type', 'text/html')
  res.send(html)
})
```

The API will simply return the generated HTML. That generated HTML is then used to generate a PDF (via headless chrome).

IMO, I do not need a client bundle since I only render the pages on the server side. Let me know if I am completely off here. Appreciate any input here.

**Comment by ankitarora05** [comment]
https://gamecom.app/

I created a website on nuxt and have a infinite loader on the feed on the landing. Gives me great performance using vue and server render. Any thoughts?

FYI you have to login to enjoy the feed :-p

**Comment by Atinux** [comment]
We have been talking with @pi0 and we could introduce `build.client` option in the `nuxt.config.js`:

```
// nuxt.config.js
export default {
  build: {
    client: false
  }
}
```

**Comment by kevcodez** [comment]
@Atinux that would be perfect!

**Comment by Timkor** [comment]
@Atinux What is the status of this? And is there some way to prevent loading the bundle client side in the mean time?

**Comment by Detzler** [comment]
@Atinux 

need this feature very much.. best I can do is to set ssr:false to skip at least the server side bundling

**Comment by juanjcardona13** [comment]
waiting for this option "client: [false",](https://github.com/nuxt/nuxt.js/issues/6860#issuecomment-577145537) time in build client is highest.
`✔ Client
  Compiled successfully in 13.24s

✔ Server
  Compiled successfully in 1.40s
`
if you  @Atinux  maybe may check again this option, It would be a great help

**Comment by Dyrio** [comment]
Is the a way to build `.output/server/` but not `.output/public/` through a `nuxt build` option ?
It could be useful. Thanks.

**Comment by kogratte** [comment]
Hey there.

Same question.

I would like to build the front part as a pure static page, but also the server side separately in order to host it behind an api manager.

What is the way to go?

**Comment by vis97c** [comment]
Is this on the way? I would love to redeploy my server routes on google cloud functions without waiting for the whole project to compile. It would also be useful if it is supported by the CLI in similar fashion to nitro presets.

**Comment by steffenstolze** [comment]
The only way I found to accomplish that is to have a nitro.config.ts file next to the nuxt.config.ts file and then run npx nitro build to only create the server part.

Creating the client without the server is hacky. One solution is to rename the server folder temporarily (also the nitro config), the run nuxt build with ssr set to false ofc. Then rename it again to the correct values - with an mjs script for example.

But as I said, feels kinda dirty.

---

I also think that this would be very helpful out of the box. For my current project, I'd like to keep the DX of Nuxt in local development but deploy the frontend as SPA to CloudFlare and run the server part in AWS Lambda. In production, CloudFlare could proxy the requests targeted at /api to the Lambda / API Gateway

**Comment by pawmanaloto** [comment]
> The only way I found to accomplish that is to have a nitro.config.ts file next to the nuxt.config.ts file and then run npx nitro build to only create the server part.
> 
> Creating the client without the server is hacky. One solution is to rename the server folder temporarily (also the nitro config), the run nuxt build with ssr set to false ofc. Then rename it again to the correct values - with an mjs script for example.
> 
> But as I said, feels kinda dirty.
> 
> I also think that this would be very helpful out of the box. For my current project, I'd like to keep the DX of Nuxt in local development but deploy the frontend as SPA to CloudFlare and run the server part in AWS Lambda. In production, CloudFlare could proxy the requests targeted at /api to the Lambda / API Gateway

Hi, trying to do the same deployment as well. Have you figure out on how to do this?
